### PR TITLE
fix: preserve Docker feature flags when enabling CDI

### PR DESCRIPTION
**Pull request by git-jxj**
## Description

Running `nvidia-ctk runtime configure --runtime=docker --enable-cdi` against a daemon.json with existing feature flags replaces them with only `cdi: true`. For example, `containerd-snapshotter: true` and `buildkit: false` disappear.

Docker configuration is decoded into `map[string]any`, but `EnableCDI` asserts the features entry as `map[string]bool`. Use the decoded map type so existing settings survive while CDI is enabled.

Add file-based tests for missing/empty features, existing true and false flags, CDI already enabled or disabled, and repeated load/update/save operations.

## Checklist

- [x] No secrets, sensitive information, or unrelated changes
- [ ] Unit tests passing (`make test`) — the full suite has the baseline failures described below.
- [x] Lint checks passing (`make lint`)
- [x] Test cases are added for new code paths
- [x] Commits are signed-off and [cryptographically signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).

## Testing

- Three preservation cases fail before the fix and pass afterward.
- `go test -race ./pkg/config/... ./cmd/nvidia-ctk/runtime/... -count=1` passed.
- `make lint`: 0 issues.
- The real CLI preserves existing features and unrelated settings in a temporary daemon.json.

`make test` has the same four failing packages and nine failing subtests on unmodified upstream and this branch: installer, installer/toolkit, modifier, and nvcdi. These involve missing test driver libraries and CSV hook expectation differences; the full suite is not reported as passing.

### CODE REVIEW

File: pkg/config/engine/docker/docker_test.go [hunk 1]
```diff
@@ -247,3 +249,33 @@ func TestGetRuntimeConfig(t *testing.T) {
...
+		{"empty features", `{"features":{}}`, `{"features":{"cdi":true}}`},
+		{"existing flags", `{"features":{"containerd-snapshotter":true,"buildkit":false},"debug":true}`, `{"features":{"cdi":true,"containerd-snapshotter":true,"buildkit":false},"debug":true}`},
+		{"CDI disabled", `{"features":{"cdi":false,"containerd-snapshotter":true}}`, `{"features":{"cdi":true,"containerd-snapshotter":true}}`},
+		{"CDI enabled", `{"features":{"cdi":true,"containerd-snapshotter":true}}`, `{"features":{"cdi":true,"containerd-snapshotter":true}}`},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			path := filepath.Join(t.TempDir(), "daemon.json")
+			require.NoError(t, os.WriteFile(path, []byte(tc.input), 0o600))
+			for i := 0; i < 2; i++ {
```
Comment (henry118) [inline pkg/config/engine/docker/docker_test.go:271]: wonder why do we need this loop?

Comment (git-jxj) [reply pkg/config/engine/docker/docker_test.go:271]: The second iteration verifies idempotency after reloading the saved file: enabling CDI again must preserve the existing features and leave the configuration unchanged. Added a comment explaining this in 545b7d3d5afadd93f902b5bb544e9b0b1a4fe7ef. `go test ./pkg/config/engine/docker -count=1` passes.

**Comment by git-jxj** [comment]
The GitHub Actions runs for current head `545b7d3d5afa` are waiting for maintainer approval (`action_required`), including [CodeQL](https://github.com/NVIDIA/nvidia-container-toolkit/actions/runs/34553456853), [Golang](https://github.com/NVIDIA/nvidia-container-toolkit/actions/runs/34553456803). Could a maintainer approve the pending workflow runs so CI can validate this head? The review explanation is in this head, and `go test ./pkg/config/engine/docker -count=1` passes locally.

**Comment by henry118** [comment]
@git-jxj Please sign commit 545b7d3d5afadd93f902b5bb544e9b0b1a4fe7ef with a GPG/SSH key. or squash two commits.

**Comment by git-jxj** [comment]
Signed the previously unsigned commit with my SSH key and pushed it as [b0ee6cde](https://github.com/NVIDIA/nvidia-container-toolkit/commit/b0ee6cde58d87cdcfa347316e0c0bd15461da899). GitHub now reports it as Verified. The code tree, parent commit, author and commit message are unchanged; `go test ./pkg/config/engine/docker -count=1` passed.

The new head still needs CI validation through the repository's vetting/approval process. Could you authorize `b0ee6cde58d87cdcfa347316e0c0bd15461da899` when convenient?

**Comment by cdesiniotis** [comment]
@git-jxj could you squash your commits to make a more readable git history? Thanks.

**Comment by git-jxj** [comment]
Squashed the two commits into [54f6ad1f](https://github.com/NVIDIA/nvidia-container-toolkit/commit/54f6ad1f4f962c1be54c11878fffe0f725bff098). The PR now contains one SSH-signed commit, which GitHub reports as Verified, with both DCO sign-offs preserved. The code tree is unchanged, including the explanation of the repeated load/enable/save test cycle.

`go test ./pkg/config/engine/docker -count=1` passes, and DCO passes. The new head’s CodeQL and Golang runs are awaiting maintainer approval (`action_required`); could you approve CI for `54f6ad1f4f962c1be54c11878fffe0f725bff098`?

File: pkg/config/engine/docker/docker_test.go (hunk 1 above)
Comment (cdesiniotis) [reply pkg/config/engine/docker/docker_test.go]: I have gone ahead and removed this for loop as it was causing a `go vet` failiure and imo it is not bringing any value.

**Review by cdesiniotis — APPROVED** [review]
